languages/java/oso/src/main/java/com/osohq/oso/Host.java

```diff
@@ -268,7 +276,21 @@ public Object toJava(JSONObject term)
                     case "Integer":
                         return num.getInt("Integer");
                     case "Float":
-                        return num.getDouble("Float");
+                        Object f = num.get("Float");
+                        if (f instanceof String) {
+                            switch ((String) f) {
+                                case "Infinity":
+                                    return Double.POSITIVE_INFINITY;
+                                case "-Infinity":
+                                    return Double.NEGATIVE_INFINITY;
+                                case "NaN":
+                                    return Double.NaN;
+                                default:
+                                    throw new Exceptions.OsoException(
+                                        "Expected a Float; received the string \"" + f + "\"");
```

Comment:
on second thought do we actually need this check since we control both ends of this serialization pipe and know this will always be one of these three strings?

Reply:
These checks are basically assertions. We "know" it will be one of the three special strings, but if it's *not*, I want to know about it.

Reply:
Word. In the spirit of keeping errors aligned across the lang libs, should we add the same assertion to Ruby and Python?

Reply:
Yes, I had that thought, and also thought, "Gabe's going to ask about the others." Thank you for not letting me ignore it.